docs/presence_routing_module.md

```diff
@@ -0,0 +1,134 @@
+# Presence Routing Module
+
+Synapse supports configuring a module that can specify additional destinations (users or
+rooms) to receive certain presence updates from local users.
+
+Both local or remote users can be specified as a destination. If a room is specified, that
+will be translated to all users in that room. The homeserver must be participating in any
+specified room.
```

Comment:
I'm not 100% convinced we want to be able to specify rooms TBH, I think it may end up leading to a bunch of complexity for not much gain (unless we have specific use cases for it)

Reply:
It was included as it would be rather easy to support ([`get_interested_parties`](https://github.com/matrix-org/synapse/blob/8bcfc2eaad66aa6e92a745d226fee4a81a921541/synapse/handlers/presence.py#L1305-L1329) already returns this), but we don't currently have any use cases. Off the top of my head it might be useful to have something like a "presence receivers" room where you could add/remove users remotely.

Reply:
Now that I've read your second comment though, supporting rooms would not be as simple.

Reply:
Well, actually the sync code already supports having an explicit room ID attached to it. We could just add more room IDs to retrieve users from :)